cmake/Modules/FindMKLDNN.cmake

```diff
@@ -1,32 +1,115 @@
 # - Try to find MKLDNN
 #
 # The following variables are optionally searched for defaults
-#  MKLDNN_ROOT_DIR:            Base directory where all MKLDNN components are found
+#  MKL_FOUND             : set to true if a library implementing the CBLAS interface is found
+#  USE_MKLDNN
 #
 # The following are set after configuration is done:
-#  MKLDNN_FOUND
-#  MKLDNN_INCLUDE_DIRS
-#  MKLDNN_LIBRARIES
-#  MKLDNN_LIBRARY_DIRS
+#  MKLDNN_FOUND          : set to true if mkl-dnn is found.
+#  MKLDNN_INCLUDE_DIR    : path to mkl-dnn include dir.
+#  MKLDNN_LIBRARIES      : list of libraries for mkl-dnn
 
-include(FindPackageHandleStandardArgs)
+IF (NOT MKLDNN_FOUND)
 
-set(MKLDNN_ROOT_DIR "" CACHE PATH "Folder contains Intel MKLDNN")
+SET(MKLDNN_LIBRARIES)
+SET(MKLDNN_INCLUDE_DIR)
 
-find_path(MKLDNN_INCLUDE_DIR mkldnn.h
-    HINTS ${MKLDNN_ROOT_DIR}
-    PATH_SUFFIXES include)
+IF (NOT USE_MKLDNN)
+  RETURN()
+ENDIF(NOT USE_MKLDNN)
 
-find_library(MKLDNN_LIBRARY mkldnn
-    HINTS ${MKLDNN_LIB_DIR} ${MKLDNN_ROOT_DIR}
-    PATH_SUFFIXES lib lib64)
+IF(MSVC)
+  MESSAGE(STATUS "MKL-DNN needs omp 3+ which is not supported in MSVC so far")
+  RETURN()
+ENDIF(MSVC)
 
-find_package_handle_standard_args(
-    MKLDNN DEFAULT_MSG MKLDNN_INCLUDE_DIR MKLDNN_LIBRARY)
+SET(IDEEP_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep")
+SET(MKLDNN_ROOT "${IDEEP_ROOT}/mkl-dnn")
 
-if(MKLDNN_FOUND)
-  set(MKLDNN_INCLUDE_DIRS ${MKLDNN_INCLUDE_DIR})
-  set(MKLDNN_LIBRARIES ${MKLDNN_LIBRARY})
-  message(STATUS "Found MKLDNN      (include: ${MKLDNN_INCLUDE_DIR}, library: ${MKLDNN_LIBRARY})")
-  mark_as_advanced(MKLDNN_ROOT_DIR MKLDNN_LIBRARY MKLDNN_INCLUDE_DIR)
-endif()
+FIND_PACKAGE(BLAS)
+FIND_PATH(IDEEP_INCLUDE_DIR ideep.hpp PATHS ${IDEEP_ROOT} PATH_SUFFIXES include)
+FIND_PATH(MKLDNN_INCLUDE_DIR mkldnn.hpp mkldnn.h PATHS ${MKLDNN_ROOT} PATH_SUFFIXES include)
+IF (NOT MKLDNN_INCLUDE_DIR)
+  EXECUTE_PROCESS(COMMAND git submodule update --init mkl-dnn WORKING_DIRECTORY ${IDEEP_ROOT})
```

Comment:
Would this have any problem on Windows?

Reply:
Do you mean the different path separator on windows and linux? It should be OK.

Reply:
Also that git might need to be `git.exe` or something in the non-Cygwin case. Maybe just wrap this in an `IF (UNIX)`?